File: 백준/Bronze/22950. 이진수 나눗셈/이진수 나눗셈.py
n = int(input())
m = input()
k = n - int(input())

t = True
if k < 0:
    k = 0

while k < n:
    if m[k] == '1':
        t = False
        break
    k += 1

print("YES" if t else "NO")

File: 백준/Bronze/22950. 이진수 나눗셈/이진수 나눗셈.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, k;
    char m[1000000];
    
    scanf("%d", &n);
    scanf("%s", m);
    scanf("%d", &k);
    k = n - k;
    
    int t = 1;
    if (k < 0) k = 0;
    
    while (k < n) {
        if (m[k] == '1') {
            t = 0;
            break;
        }
        k++;
    }
    
    printf("%s\n", t ? "YES" : "NO");
    
    return 0;
}
